sklearn/preprocessing/data.py

```diff
@@ -137,7 +137,7 @@ def scale(X, axis=0, with_mean=True, with_std=True, copy=True):
 
     """  # noqa
     X = check_array(X, accept_sparse='csc', copy=copy, ensure_2d=False,
-                    warn_on_dtype=True, estimator='the scale function',
+                    warn_on_dtype=False, estimator='the scale function',
```

Comment:
can also just remove the `warn_on_dtype=False` here, as that is the default of `check_array`